astro/tvm/utils.go

```diff
@@ -67,6 +69,9 @@ func unzip(zipfilePath string, destDir string) error {
 		defer fh.Close()
 
 		path := filepath.Join(destDir, f.Name)
+		if !strings.HasPrefix(path, filepath.Clean(destDir)+string(os.PathSeparator)) {
```

Comment:
This is good, however, as others pointed out in https://github.com/golang/go/issues/25849, the check can still bypassed by symlinks. Although the symlink is implemented in tar and not in Zip, so it should be okay.

Reply:
A small note: This relies on `Join` performing a `Clean` on the path. Imagine if that doesn't happen any longer, then the check will be basically useless. For example:
```
path := "/home/friendly/../../etc/shadow"
strings.HasPrefix(path, "/home/friendly/") == true
```

An explicit Clean before comparison wouldn't harm.

Reply:
Since that would be an observable change to the `filepath.Join` api (it not calling Clean), the chances of that ever happening are next to 0. It's not just observable, it's actually documented `Join calls Clean on the result` say [the docs](https://golang.org/pkg/path/filepath/#Join), so it'd actually be a violation of the Go 1.x compatibility contract to change it.

Reply:
Agree with @jcorbin, gonna leave it